### PR TITLE
boards: arm: mark nRF5340 PDK as deprecated.

### PR DESCRIPTION
**Pull request by ioannisg**
We deprecate nRF5340 PDK and add a note that
the board will be replaced by nRF5340 DK.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Comment by ioannisg** [comment]
Note: Board documentation still points to the PDK, as the links to the nRF5340 DK are not yet available. Please advise otherwise.

**Review by tejlmand — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/doc/index.rst [hunk 1]
```diff
@@ -54,7 +54,9 @@ contains the processor's information and the datasheet.
 
 .. note::
 
-   In previous Zephyr releases this board was named *nrf5340_dk_nrf5340*.
+   * nRF5340 PDK board is deprecated and will be replaced by nRF5340 DK.
+     Support for nRF5340 DK will be removed in Zephyr v2.6.0 release.
```
Comment (anangl) [inline boards/arm/nrf5340dk_nrf5340/doc/index.rst:58]:
```suggestion
     Support for nRF5340 PDK will be removed in Zephyr v2.6.0 release.
```

**Review by anangl — CHANGES_REQUESTED** [review]

File: boards/arm/nrf5340dk_nrf5340/doc/index.rst (hunk 1 above)
Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/doc/index.rst:58]: ups, thanks @anangl :) :)

**Review by anangl — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/doc/index.rst [hunk 2]
```diff
@@ -54,7 +54,9 @@ contains the processor's information and the datasheet.
 
 .. note::
 
-   In previous Zephyr releases this board was named *nrf5340_dk_nrf5340*.
+   * nRF5340 PDK board is deprecated and will be replaced by nRF5340 DK.
```
Comment (ru-fu) [inline boards/arm/nrf5340dk_nrf5340/doc/index.rst:57]: No need for a bullet list here, but doesn't hurt either.

**Review by ru-fu — APPROVED** [review]

File: boards/arm/nrf5340dk_nrf5340/doc/index.rst (hunk 2 above)
Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/doc/index.rst:57]: Agreed, this was copy-paste mistake. Will remove ):)

Comment (ioannisg) [reply boards/arm/nrf5340dk_nrf5340/doc/index.rst]: removed.

**Pull request merged**